src/api/channel.rs

```diff
@@ -0,0 +1,649 @@
+// Copyright (c) 2018-2019, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+#![allow(missing_docs)]
+
+use crate::api::color::*;
+use crate::api::config::*;
+use crate::api::internal::ContextInner;
+use crate::api::util::*;
+
+use bitstream_io::*;
+use crossbeam::channel::*;
+
+use crate::api::context::RcData;
+use crate::encoder::*;
+use crate::frame::*;
+use crate::rate::RCState;
+use crate::rayon::{ThreadPool, ThreadPoolBuilder};
+use crate::util::Pixel;
+
+use std::io;
+use std::sync::Arc;
+
+/// Endpoint to send previous-pass statistics data
+pub struct RcDataSender {
+  sender: Sender<RcData>,
```

Comment:
Could we use the `bounded` helper in crossbeam to avoid having to manually implement senders and receivers? It seems a lot of this code is boilerplate and would be provided by crossbeam.

Reply:
No since those endpoint need to do more than that: 
- error reporting
- `new_frames()`
- `container_sequence_header()`